utils/python/CIME/XML/env_mach_pes.py

```diff
@@ -95,6 +95,8 @@ def get_total_tasks(self, comp_classes):
             pstrid = self.get_value("PSTRID_%s"%comp)
             tt = rootpe + (ntasks - 1) * pstrid + 1
             total_tasks = max(tt, total_tasks)
+        if total_tasks < self.get_value("PES_PER_NODE"): # all PEs fit in 1 node
```

Comment:
I think that we can add code in env_mach_specific to assure that the value applied to -n is >= that applied to -N  .     I can help you figure that out if you find it a suitable alternative.